bindings/cs/Scripts/Create-and-pack-nuget.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+# This script creates 4 folder inside $packagingFolder
+# 1. libmongocrypt-all. This folder contains all platform-specific C libraries.
+# 2. csharp-v{$bindingsReleaseVersion}. After all, this folder will containts the fully ready source code.
+# 3. downloaded_mongocrypt. This folder contains the used in the libmongocrypt platform-specific C libraries.
+# 4. artifacts. After all, the generated .nupkg files will be placed here.
+#
+# Input data:
+# $libmongocryptAllUrl - this is the link on EG patch that contains all required platform-specific C libraries.
+
+packagingFolder="c:/build"
+bindingsReleaseVersion="1.2.1"
```

Comment:
Any variables that change version to version (e.g. `bindingsReleaseVersion`, `branchName`, and `libmonogcryptAllUrl`) should be required parameters rather than hard-coded values.

Reply:
For `libmongocryptAllUrl`, we might be able to use the `latest` URL, which should point to the latest successful build. We should confirm this will always contain the desired unmanaged dependencies. If so, we can hard-code this URL in the script.
https://mciuploads.s3.amazonaws.com/libmongocrypt/all/master/latest/libmongocrypt-all.tar.gz

Reply:
agreed about the first. As for the second, I think we cannot rely on the above link. For example the latest libmongocrypt release, for now, is `1.2.1`, but we changed the code only for `1.2.0`. Idk whether it's problematic to use the latest library in the next release without our code changes, but even if not so, we can have such problems later, but I will confirm this understanding

Reply:
My intent was to avoid having to specify a SHA1 hash as a command line argument. Maybe ask if the official `libmongocrypt` builds could be symlinked to the SHA1 hashes automatically. For example:
https://mciuploads.s3.amazonaws.com/libmongocrypt/all/v1.2.0/libmongocrypt-all.tar.gz

would be symlinked to:

https://mciuploads.s3.amazonaws.com/libmongocrypt/all/master/508e21f4abff9f5519e0357a63a4ad30d2c24692/libmongocrypt-all.tar.gz

I would rather remember a tag name than a SHA1. Plus tag names are easier to visually validate than a SHA1. (e.g. Did I copy the entire value? Did I leave off a digit?)

Reply:
yeah, I will check what we can do here

Reply:
Kevin is out this week, I think we can move this logic to a separate ticket. I will discuss it with him when he will be back

Reply:
> Idk whether it's problematic to use the latest library in the next release without our code changes, but even if not so, we can have such problems later, but I will confirm this understanding

Ideally it shouldn't. If in the unlikely event we do a backwards breaking release, we'd have to ensure drivers opt-in to consuming it. But, specifying the tag guarantees you don't get an unwanted upgrade.

I agree having a URL available via a tag seems like a good idea. I filed https://jira.mongodb.org/browse/MONGOCRYPT-315 to make that change.